url49584c1d41_Arrem_GGS_tgz-pJ8-com_gamezgalaxy_test_console_MainJ8/src/com/gamezgalaxy/GGS/util/properties/Properties.java

```diff
@@ -0,0 +1,268 @@
+import java.io.PrintWriter;
+import java.util.ArrayList;
+
+import com.gamezgalaxy.GGS.server.Server;
+
+public class Properties {
+	private ArrayList<String> settings = new ArrayList<String>();
+	private static boolean init = false;
+	
+	/**
+	 * Load server settings
+	 * <b>This method can only be called once</b>
+	 * @param server
+	 *              The server to load for
+	 * @return
+	 *        The {@link Properties} object
+	 */
+	public static Properties init(Server server) {
+		if (init)
+			return null;
+		Properties p = new Properties();
+		if (!new File("properties").exists())
+			new File("properties").mkdir();
+		if (!new File("properties/" + server.configpath).exists())
+			makeDefaults(server.configpath, server, p);
+		else {
+			try {
+				p.load(server.configpath);
+				init = true;
+			} catch (IOException e) {
+				server.Log("ERROR LOADING CONFIG!");
+				e.printStackTrace();
+			}
+		}
+		return p;
+		
+	}
+	
+	private static void makeDefaults(String filename, Server server, Properties p) {
+		//TODO Fill in all defaults
+		server.Log("System config not found..creating..");
+		p.addSetting("Server-Name", "[GGS] Default Server");
+		p.addSetting("WOM-Alternate-Name", "[GGS] Default Server");
+		p.addSetting("MOTD", "Welcome!");
+		p.addSetting("Port", 25565);
+		p.addSetting("Max-Players", 30);
+		p.addSetting("Public", true);
+		p.addSetting("WOM-Server-description", "A server");
+		p.addSetting("WOM-Server-Flags", "[GGS]");
+		p.addSetting("MainLevel", "levels/Main.ggs");
+		p.addSetting("SQL-Driver", "com.gamezgalaxy.GGS.sql.SQLite");
+		p.addSetting("SQL-table-prefix", "ggs");
+		p.addSetting("MySQL-username", "root");
+		p.addSetting("MySQL-password", "password");
+		p.addSetting("MySQL-database-name", "ggsdb");
+		try {
+			p.save(filename);
+			init = true;
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Save the properties file
+	 * @param filename
+	 *                The file to save it as
+	 * @throws IOException
+	 *                    This is thrown if there's an error saving the file
+	 */
+	public void save(String filename) throws IOException {
+		if (new File("properties/" + filename).exists())
+			new File("properties/" + filename).delete();
+		new File("properties/" + filename).createNewFile();
+		PrintWriter out = new PrintWriter("properties/" + filename);
+		for (String s : settings) {
+			out.println(s);
+		}
+		out.close();
+	}
+	
+	/**
+	 * Load the properties file. 
+	 * This will reload the settings in memory.
+	 * @param filename
+	 *                The name of the file
+	 * @throws IOException
+	 *                     This is thrown if there's an error reading the file
+	 */
+	public void load(String filename) throws IOException {
+		if (settings.size() > 0)
+			settings.clear();
+		FileInputStream fstream = new FileInputStream("properties/" + filename);
+		DataInputStream in = new DataInputStream(fstream);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *in*. The resource is referenced at lines 112, 113, 115. The resource is closed at line 120. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).